doc/man3/SSL_set_retry_verify.pod

```diff
@@ -0,0 +1,69 @@
+=pod
+
+=head1 NAME
+
+SSL_set_retry_verify - indicate that certificate verification should be retried
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ int SSL_set_retry_verify(SSL *ssl);
+
+=head1 DESCRIPTION
+
+SSL_set_retry_verify() should be called from the certificate verification
+callback on a client when the application wants to indicate that the handshake
+should be suspended and the control should be returned to the application.
+L<SSL_want_retry_verify(3)> will return 1 as a consequence.
+
+Please refer to L<SSL_CTX_set_cert_verify_callback(3)> for further details.
+
+=head1 NOTES
+
+The effect of calling SSL_set_retry_verify() outside of the certificate
+verification callback on the client side is undefined.
+
+=head1 RETURN VALUES
+
+SSL_set_retry verify() returns 1 on success, 0 otherwise.
+
+=head1 EXAMPLES
+
+The following code snippet shows how obtain the B<SSL> object associated with
+the B<X509_STORE_CTX> to call the SSL_set_retry_verify() function:
+
+    int idx = SSL_get_ex_data_X509_STORE_CTX_idx();
+    SSL *ssl;
+
+    /* this should not happen but check anyway */
+    if (idx < 0
+        || (ssl = X509_STORE_CTX_get_ex_data(ctx, idx)) == NULL) 
+        return 0;
+
+    if (/* we need to retry verification callback */)
+        return SSL_set_retry_verify(ssl);
```

Comment:
I suggest putting all the code shown here under the condition
`if (/* we need to retry verification callback */)`

Reply:
Again, all this could be simplified a lot avoid all the new local vars,
if you make `SSL_set_retry_verify()` return `0` on its input being `NULL`.

As mentioned, we do have the freedom here because `SSL_CTRL_SET_RETRY_VERIFY` is a new case.
Or what else do you see speak against this?

Reply:
In my opinion although this simplifies the application code in case of simple apps, other applications should really know what they are doing. Those calls can fail but the failure is outside the SSL_set_retry_verify() call and should not be masked by calling that function anyway with NULL SSL parameter. So no, I am not going to do that.

Reply:
If the fetching of `ssl` fails for whatever reason, `SSL_set_retry_verify()` would propagate this automatically.
Anyway, we do not need to finish this discussion as part of this PR (I'm going to suggest a fixup PR).